src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdNewCaseCreator.java

```diff
@@ -183,24 +191,44 @@ private long createNewCaseInCcd(
     private CaseDataContent getCaseDataContent(
         Map<String, Object> caseData,
         String exceptionRecordId,
-        String eventId,
-        String eventToken
+        String envelopeId,
+        StartEventResponse startEventResponse,
+        String service
     ) {
-        Map<String, Object> data = new HashMap<>(caseData);
-        data.put(EXCEPTION_RECORD_REFERENCE, exceptionRecordId);
+        Map<String, Object> completeCaseData =
+            setBulkScanSpecificFieldsInCaseData(caseData, service, exceptionRecordId, envelopeId);
 
         return CaseDataContent
             .builder()
             .caseReference(exceptionRecordId)
-            .data(data)
+            .data(completeCaseData)
             .event(Event
                 .builder()
-                .id(eventId)
+                .id(startEventResponse.getEventId())
                 .summary("Case created")
                 .description("Case created from exception record ref " + exceptionRecordId)
                 .build()
             )
-            .eventToken(eventToken)
+            .eventToken(startEventResponse.getToken())
             .build();
     }
+
+    private Map<String, Object> setBulkScanSpecificFieldsInCaseData(
+        Map<String, Object> caseData,
+        String service,
+        String exceptionRecordId,
+        String envelopeId
+    ) {
+        Map<String, Object> updatedCaseData = new HashMap<>(caseData);
+        updatedCaseData.put(BULK_SCAN_CASE_REFERENCE, exceptionRecordId);
+
+        if (envelopeReferenceCollectionHelper.serviceSupportsEnvelopeReferences(service)) {
```

Comment:
can we add some logging here and also for else part ?

Reply:
done